polynote-kernel/src/test/scala/polynote/kernel/lang/KernelSpec.scala

```diff
@@ -28,47 +29,37 @@ trait KernelSpec {
   implicit val contextShift: ContextShift[IO] = IOContextShift(ExecutionContext.fromExecutorService(Executors.newCachedThreadPool()))
 
   def assertPythonOutput(code: String)(assertion: (Map[String, Any], Seq[Result], Seq[(String, String)]) => Unit): Unit = {
-    assertOutput((rst: RuntimeSymbolTable) => PythonInterpreter.factory()(Nil, rst), code)(assertion)
+    assertOutput((ctx: RuntimeContext) => PythonInterpreter.factory()(Nil, ctx), code)(assertion)
   }
 
   def assertScalaOutput(code: String)(assertion: (Map[String, Any], Seq[Result], Seq[(String, String)]) => Unit): Unit = {
-    assertOutput((rst: RuntimeSymbolTable) => ScalaInterpreter.factory()(Nil, rst), code)(assertion)
+    assertOutput((ctx: RuntimeContext) => ScalaInterpreter.factory()(Nil, ctx), code)(assertion)
   }
 
   // TODO: for unit tests we'd ideally want to hook directly to runCode without needing all this!
-  def assertOutput(mkInterp: RuntimeSymbolTable => LanguageKernel[IO], code: String)(assertion: (Map[String, Any], Seq[Result], Seq[(String, String)]) => Unit): Unit = {
+  def assertOutput(mkInterp: RuntimeContext => LanguageKernel[IO], code: String)(assertion: (Map[String, Any], Seq[Result], Seq[(String, String)]) => Unit): Unit = {
     Topic[IO, KernelStatusUpdate](UpdatedTasks(Nil)).flatMap { updates =>
-      val symbolTable = new RuntimeSymbolTable(GlobalInfo(Map.empty, Nil), updates)
-      val interp = mkInterp(symbolTable)
-      interp.init().unsafeRunSync()
+      val runtimeCtx = new RuntimeContext(GlobalInfo(Map.empty, Nil))
+      val interp = mkInterp(runtimeCtx)
+
       val displayed = mutable.ArrayBuffer.empty[(String, String)]
       polynote.runtime.Runtime.setDisplayer((mimeType, input) => displayed.append((mimeType, input)))
 
       // TODO: we should get the results of out as well so we can capture output (or maybe interpreters shouldn't even be writing to out!!)
-      Queue.unbounded[IO, Option[Result]].flatMap {
-        out =>
-          val oqSome = new EnqueueSome(out)
-          Topic[IO, KernelStatusUpdate](UpdatedTasks(Nil)).flatMap {
-            statusUpdates =>
-              for {
-                // publishes to symbol table as a side-effect
-                // TODO: ideally we wouldn't need to run predef specially
-                _ <- interp.runCode("test", symbolTable.currentTerms.map(_.asInstanceOf[interp.Decl]), Nil, code, oqSome, statusUpdates)
-                // make sure everything has been processed
-                _ <- symbolTable.drain()
-                _ <- out.enqueue1(None) // terminate the queue
-                // these are the vars runCode published
-                vars = symbolTable.currentTerms
-              } yield {
-                val namedVars = vars.map(v => v.name.toString -> v.value).toMap
+      for {
```

Comment:
sorry, I introduced some conflicts for you here 😞

Reply:
no worries! Resolving them wasn't bad at all, pushed :)